psycopg/psycopg/_typeinfo.py

```diff
@@ -154,7 +154,7 @@ def _get_info_query(
     typname AS name, oid, typarray AS array_oid,
     oid::regtype::text AS regtype, typdelim AS delimiter
 FROM pg_type t
-WHERE t.oid = %(name)s::regtype
+WHERE t.oid = to_regclass(%(name)s)
```

Comment:
Together with these changes, you want to drop the `try... except UndefinedObject` where the query is called, because the function, unlike the cast, is not supposed to raise it.